src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1012,6 +1048,9 @@ private void AddTransactionToWallet(Transaction transaction, TxOut utxo, int? bl
                 {
                     // Figure out how to retrieve the destination address.
                     string destinationAddress = this.scriptAddressReader.GetAddressFromScriptPubKey(this.network, paidToOutput.ScriptPubKey);
+                    if (destinationAddress == string.Empty)
```

Comment:
Do you see a case where we can't get the address?

Reply:
Cold staking scripts, being associated with two public keys, do not have a single deterministic address that can be derived from the keys alone. However, in the wallet context, we are tracking a maximum of one wallet address per script hence we can use that lookup to resolve an address. Leveraging the lookup in this way should be a generic.